src/enrichment_tables/pgtable.rs

```diff
@@ -0,0 +1,521 @@
+        match index {
+            None => {
+                // No index has been passed so we need to do a Sequential Scan.
+                Ok(self
+                    .sequential(self.data.iter(), case, condition, select)
+                    .collect())
+            }
+            Some(handle) => {
+                // Perform a sequential scan over the indexed result.
+                Ok(self
+                    .sequential(
+                        self.indexed(case, condition, handle)?
+                            .iter()
+                            .flat_map(|results| results.iter().map(|idx| &self.data[*idx])),
+                        case,
+                        condition,
+                        select,
+                    )
+                    .collect())
+            }
+        }
+    }
+
+    /// Returns a list of the field names that are in each index
+    fn index_fields(&self) -> Vec<(Case, Vec<String>)> {
+        self.indexes
+            .iter()
+            .map(|index| {
+                let (case, fields, _) = index;
+                (
+                    *case,
+                    fields
+                        .iter()
+                        .map(|idx| self.headers[*idx].clone())
+                        .collect::<Vec<_>>(),
+                )
+            })
+            .collect::<Vec<_>>()
+    }
+
+    fn add_index(&mut self, case: Case, fields: &[&str]) -> Result<IndexHandle, String> {
+        let normalized = self.normalize_index_fields(fields)?;
+        match self
+            .indexes
+            .iter()
+            .position(|index| index.0 == case && index.1 == normalized)
+        {
+            Some(pos) => {
+                // This index already exists
+                Ok(IndexHandle(pos))
+            }
+            None => {
+                let index = self.index_data(&normalized, case)?;
+                self.indexes.push((case, normalized, index));
+                // The returned index handle is the position of the index in our list of indexes.
+                Ok(IndexHandle(self.indexes.len() - 1))
+            }
+        }
+    }
+
+    /// Checks the modified timestamp of the data file to see if data has changed.
+    fn needs_reload(&self) -> bool {
+        let old = self.updated;
+        let new = match &self.statement {
+            Some(s) => match block_on(self.client.query_opt(s,&[])) {
+                Ok(res) => match res {
+                    Some(row) => match row.try_get(0) {
+                        Ok(t) => { t }
+                        Err(e) => { return true } // On error, request a reload? Not sure what to do here.
+                    }
+                    None => { DateTime::UNIX_EPOCH }
+                }
+                Err(e) => { return true } // On error, request a reload? Not sure what to do here.
+            }
+            None => { DateTime::UNIX_EPOCH }
+        };
+        if new > old {
+            self.updated = new;
```

Comment:
This isn't necessary. If the table needs reloading the whole struct is discarded and a new one is created. `updated` will need to be set when the table is loaded.